python_modules/dagster/dagster_tests/core_tests/storage_tests/test_fs_io_manager.py

```diff
@@ -360,3 +365,57 @@ def four(inp):
         assert os.path.isfile(filepath_b)
         with open(filepath_b, "rb") as read_obj:
             assert pickle.load(read_obj) == 4
+
+
+def test_fs_io_manager_none():
+    with tempfile.TemporaryDirectory() as tmpdir_path:
+        io_manager_def = fs_io_manager.configured({"base_dir": tmpdir_path})
+
+        @asset
+        def asset1() -> None:
+            pass
+
+        @asset(non_argument_deps={"asset1"})
+        def asset2() -> None:
+            pass
+
+        result = materialize(
+            with_resources([asset1, asset2], resource_defs={"io_manager": io_manager_def})
+        )
+
+        assert not os.path.exists(os.path.join(tmpdir_path, "asset1"))
+        assert not os.path.exists(os.path.join(tmpdir_path, "asset2"))
+        handled_output_events = list(
+            filter(lambda evt: evt.is_handled_output, result.all_node_events)
+        )
+        assert len(handled_output_events) == 2
+
+        for event in handled_output_events:
+            assert len(event.event_specific_data.metadata_entries) == 0
+
+
```

Comment:
should we add a test to protect / demonstrate the behavior around being able to handle `None` in a custom io manager?

Reply:
agreed - there should be a test in the generalized io manager tests that documents the behavior of `Nothing` with `handle_output`

Reply:
added this